src/main/java/com/github/caldav4j/CalDAVCollection.java

```diff
@@ -76,11 +76,17 @@
  * no customized queries should be public in this class
  */
 public class CalDAVCollection extends CalDAVCalendarCollectionBase{
+    
 	private static final Logger log = LoggerFactory.getLogger(CalDAVCollection.class); 
-	// configuration settings
+	
+	private final ResponseToResource mapper;
 
+	protected CalDAVCollection(ResponseToResource mapper) {
```

Comment:
Question: can we define a new CardDAVCollection class to avoid changing the interface?